spec/requests/harvest/harvest_batch_benchmark_spec.rb

```diff
@@ -76,20 +76,29 @@
       expect_enqueued_jobs(duplicates + 1, of_class: BawWorkers::Jobs::Harvest::HarvestJob)
     end
 
+    # performs slighly better than `duplicate_rows_but_with_nonexistent_path`
+    # approach in `harvest_batch_spec.rb`
     def duplicate_rows
-      columns = (HarvestItem.column_names - ['id']).join(', ')
-
-      duplicate_rows_query = <<~SQL
-        INSERT INTO harvest_items (#{columns})
-        (
-          SELECT  #{columns}
-          FROM  (
-            SELECT #{columns} FROM harvest_items ORDER BY id DESC LIMIT 1
-          ) AS t
-          CROSS JOIN LATERAL generate_series(1,#{duplicates})
-        )
-      SQL
-      ActiveRecord::Base.connection.execute(duplicate_rows_query)
+      columns = (HarvestItem.column_names - ['id', 'path', 'info', 'audio_recording_id']).join(', ')
+      values = ActiveRecord::Base.connection.execute("SELECT #{columns} FROM harvest_items LIMIT 1").to_a[0].values
+      paths = duplicates.times.map { |i|
+        generate_recording_name(
+          Time.new(2021, 10, 12, 13, 24, 57, 'Z'), suffix: i.to_s, ambiguous: true, extension: '.ogg'
+        ).tap { |n| copy_fixture_to_harvest_directory(Fixtures.audio_file_mono, harvest, target_name: n) }
+      }
+      values_list = duplicates.times.map { |i|
+        "(#{[ActiveRecord::Base.connection.quote(paths[i]),
+             ActiveRecord::Base.connection.quote({ error: 'null' }.to_json), 'NULL',
+             values.map { |v| ActiveRecord::Base.connection.quote(v) }].join(', ')})"
+      }.join(', ')
+
+      ActiveRecord::Base.connection.execute(
+        <<~SQL.squish
+          INSERT INTO harvest_items
+          (path, info, audio_recording_id, #{columns})
+          VALUES #{values_list}
+        SQL
+      )
```

Comment:
maybe it's late, but i'm finding this really hard to parse.

My instinct again, is to revert your changes, and to proceed with a smaller changeset. 

Why does the performance matter? I doubt it is measurably faster.

Why is there so much extra generation?

Does the non-duplicated paths even need to exist?